src/twisted/internet/_newtls.py

```diff
@@ -159,7 +159,13 @@ def startTLS(transport, contextFactory, normal, bypass):
 
 
 @implementer(ITLSTransport)
-class ConnectionMixin(object):
+class ConnectionMixin(object):  # type: ignore[misc]
```

Comment:
Since `ConnectionMixin` is defined in `_newtls.py`, maybe we can assume that entire module is private?
If so, would it be better to remove @implementer(ITLSTransport) from the ConnectionMixin, and add it to `twisted.internet.tcp.Connection`?

I really don't like how some of these interfaces have been defined and inherit from each other....

Reply:
I've created a new PR for this: https://github.com/twisted/twisted/pull/1299

and removed the changes to ConnectionMixin from here.